app/models/ems_refresh/save_inventory_helper.rb

```diff
@@ -46,7 +46,7 @@ def save_inventory_multi(association, hashes, deletes, find_key, child_keys = []
     unless deletes.blank?
       type = association.proxy_association.reflection.name
       _log.info("[#{type}] Deleting #{log_format_deletes(deletes)}")
-      association.delete(deletes)
+      disconnect ? deletes.map(&:disconnect_inv) : association.delete(deletes)
```

Comment:
I am very hesitant about adding this to save_inventory_multi.  Also, I don't think you want `.map` here...you want `.each`